fix: Fixes various issue in the web app deployment script

ltwlf opened the pull request:
## Description
The azure webapp deployment script has various issues e.g. bf luis:build has wrong --dialog argument.
appsettings.deployment.json will never be used by the published bot and exposes local secrets to the remote folder.

## Task Item
fixes #4049

ltwlf commented on runtime/dotnet/azurewebapp/Scripts/deploy.ps1:
```diff
@@ -63,37 +63,46 @@ dotnet user-secrets init
-
 if (-not $botPath) {
 	# If don't provide bot path, then try to copy all dialogs except the runtime folder in parent folder to the publishing folder (bin\Realse\ Folder)
-	$botPath = '..'
```
Wrong default path. Must be on level higher.

ltwlf commented on runtime/dotnet/azurewebapp/Scripts/deploy.ps1:
```diff
@@ -63,37 +63,46 @@ dotnet user-secrets init
 Copy-Item -Path (Get-Item -Path $botPath -Exclude ('runtime', 'generated')).FullName -Destination $remoteBotPath -Recurse -Force -Container
 
 # Try to get luis config from appsettings
-$settings = Get-Content $(Join-Path $projFolder appsettings.deployment.json) | ConvertFrom-Json
```
appsettings.deployment.json has no effect to the published bot, should be appsettings.json

ltwlf commented on runtime/dotnet/azurewebapp/Scripts/deploy.ps1:
```diff
@@ -63,37 +63,46 @@ dotnet user-secrets init
 	$luisAuthoringKey = $luisSettings.authoringKey
 }
 
-if (-not $luisEndpointKey) {
```
Secrets/key should not be published to the remote host -> should be configured via azure setting: luis__endpointkey

ltwlf commented on runtime/dotnet/azurewebapp/Scripts/deploy.ps1:
```diff
@@ -63,37 +63,46 @@ dotnet user-secrets init
 	$luisAuthoringRegion = $luisSettings.region
 }
 
+# set feature configuration
```
Use the local settings features if configured

ltwlf commented on runtime/dotnet/azurewebapp/Scripts/deploy.ps1:
```diff
@@ -115,49 +124,40 @@ if ($luisAuthoringKey -and $luisAuthoringRegion) {
 			New-Item -ItemType Directory -Force -Path generated
 		}
-		
-		bf luis:build --luConfig $(Join-Path $remoteBotPath luconfig.json) --botName $name --authoringKey $luisAuthoringKey --dialog --out .\generated --suffix $customizedEnv -f --region $luisAuthoringRegion
```
the argument --dialog without value causes an error. In the latest branch it mus be crosstrained

ltwlf commented on runtime/dotnet/azurewebapp/Scripts/deploy.ps1:
```diff
@@ -115,49 +124,40 @@ if ($luisAuthoringKey -and $luisAuthoringRegion) {
+		bf luis:build --luConfig $(Join-Path $remoteBotPath luconfig.json) --botName $name --authoringKey $luisAuthoringKey --dialog crosstrained  --out .\generated --suffix $environment -f --region $luisAuthoringRegion
 	}
 	else {
 		Write-Host "bf luis:build does not exist, use the following command to install:"
```
Meanwhile there is a published version of the CLI

ltwlf commented on runtime/dotnet/azurewebapp/Scripts/deploy.ps1:
```diff
@@ -226,27 +225,12 @@ if ($luisAuthoringKey -and $luisAuthoringRegion) {
-
 $settings | Add-Member -Type NoteProperty -Force -Name 'feature' -Value $featureConfig
-$settings | ConvertTo-Json -depth 100 | Out-File $(Join-Path $publishFolder appsettings.deployment.json)
+$settings | ConvertTo-Json -depth 100 | Out-File $settingsPath
```
Replace settings in composer settings folder. The settings will only includes the luis app id mappings and feature configuration. All other settings must be configured via the azure web app settings

luhan2017 commented:
@ltwlf , we are deprecating this scripts, and use the scripts generated in your bot folder instead, could you please try that one? sorry for the confusion.

ltwlf commented:
I optimized the deploy script for CI/CD deployment and updated the readme accordingly. And I deleted the obsolete scripts. As soon this is merged I'm going to record a screen cast on how to do CI/CD pipelines  with composer projects.
@VanyLaw can you review the PR?

cwhitten commented:
@ltwlf please update/rebase your branches to represent the appropriate diff. Many commits outside this change are now included and the surface area is much too large.

ltwlf commented:
@cwhitten I rebased on the current main. The file changes are now correct again.
